src/Widgets/Presets/DefaultSingleitemPreset.php

```diff
@@ -0,0 +1,252 @@
+        $text .= $this->getShopBuilderDataFieldProvider('VariationGlobalDataFieldProvider::number',array('variation.number'));
+
+        $this->stickyContainer->createChild('sticky', 'Ceres::InlineTextWidget')
+            ->withSetting('appearance', 'none')
+            ->withSetting('customClass','text-muted')
+            ->withSetting('text', $text)
+            ->withSetting('spacing.customPadding', true)
+            ->withSetting('spacing.padding.left.value', 0)
+            ->withSetting('spacing.padding.left.unit', null)
+            ->withSetting('spacing.padding.right.value', 0)
+            ->withSetting('spacing.padding.right.unit', null)
+            ->withSetting('spacing.padding.top.value', 2)
+            ->withSetting('spacing.padding.top.unit', null)
+            ->withSetting('spacing.padding.bottom.value', 0)
+            ->withSetting('spacing.padding.bottom.unit', null);
+    }
+
+    private function createItemImageWidget()
+    {
+        $this->twoColumnWidget->createChild('first', 'Ceres::ItemImageWidget')
+            ->withSetting('appearance', 'primary')
+            ->withSetting('maxQuantity', 10)
+            ->withSetting('imageSize', 'urlMiddle')
+            ->withSetting('showThumbs', true)
+            ->withSetting('showDots', true);
+    }
+
+    private function createSeparatorWidget()
+    {
+        $this->stickyContainer->createChild('sticky', 'Ceres::SeparatorWidget')
+            ->withSetting('customMargin', true)
+            ->withSetting('margin.top.value', 2)
+            ->withSetting('margin.top.unit', null)
+            ->withSetting('margin.bottom.value', 2)
+            ->withSetting('margin.bottom.unit', null);
+    }
+
+    private function createLegalInformation()
+    {
+        $text ="* {% if services.customer.showNetPrices() %}{{ trans(\"Ceres::Template.singleItemExclVAT\") }}{% else %}{{ trans(\"Ceres::Template.singleItemInclVAT\") }}{% endif %} {{ trans(\"Ceres::Template.singleItemExclusive\") }}";
+        $text .="<a {% if ceresConfig.global.shippingCostsCategoryId > 0 %} data-toggle=\"modal\" href=\"#shippingscosts\"{% endif %} title=\"{{ trans(\"Ceres::Template.singleItemShippingCosts\") }}\">{{ trans(\"Ceres::Template.singleItemShippingCosts\") }}</a>";
+        $this->stickyContainer->createChild('sticky', 'Ceres::CodeWidget')
+            ->withSetting('text', $text)
+            ->withSetting('appearance', 'none');
+    }
+
+    private function createGraduatedPriceWidget()
+    {
+        $this->stickyContainer->createChild('sticky', 'Ceres::GraduatedPriceWidget')
+            ->withSetting('appearance', 'primary');
+    }
+
+    private function createItemBundleWidget()
+    {
+        $this->stickyContainer->createChild('sticky', 'Ceres::ItemBundleWidget')
+            ->withSetting('appearance', 'primary');
+    }
+
+    private function createTabWidget()
+    {
+        $uuidTabDescription = '3a0ca715-ff40-4446-8393-07f663ce45a2';
```

Comment:
hier könnte wahrscheinlich die PHP funktion "uniqid" benutzt werden, klär das am besten mit einem PHP-ler noch mal ab. Das wird in einer TWIG-Extension bereits genutzt:
https://github.com/plentymarkets/plugin-io/blob/stable/src/Extensions/Functions/UniqueId.php#L28

Reply:
Jo nach so etwas hab ich gesucht ist eingebaut 👍

Reply:
Und wo ist es?